src/ApiManagement/ApiManagement.ServiceManagement/Commands/GetAzureApiManagementGateway.cs

```diff
@@ -0,0 +1,60 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using System;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApiManagementGateway", DefaultParameterSetName = GetAll, SupportsShouldProcess = true)]
+    [OutputType(typeof(PsApiManagementGateway))]
+    public class GetAzureApiManagementGateway : AzureApiManagementCmdletBase
+    {
+        private const string GetAll = "GetAllGateways";
+        private const string GetById = "GetByGatewayId";
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Instance of PsApiManagementContext. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementContext Context { get; set; }
+
+        [Parameter(
+            ParameterSetName = GetById,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Identifier of a gateway. If specified will try to find gateway by the identifier. This parameter is optional.")]
```

Comment:
please remove "This parameter is optional" it could be misleading.

Reply:
Done